src/Aquifer.Data/DapperExtensions.cs

```diff
@@ -407,6 +407,43 @@ public static async Task<IEnumerable<TReturn>> QueryWithRetriesAsync<TFirst, TSe
             splitOn));
     }
 
+    /// <summary>
+    /// Creates a Dapper TVP from the given <paramref name="source"/> where the properties of <typeref name="T"/>
+    /// correspond to the SQL column names.
+    /// </summary>
+    /// <typeparam name="T">The type corresponding to the SQL object's schema.</typeparam>
+    /// <param name="source">The items to convert.</param>
+    /// <param name="typeName">Database type name.</param>
+    /// <returns>Table-Valued Parameter</returns>
+    public static SqlMapper.ICustomQueryParameter AsTableValuedParameter<T>(this IEnumerable<T> source, string typeName)
```

Comment:
Dapper only supports creation of TVPs from a `DataTable`.  This method allows passing any `IEnumerable` (note that it comes with a minor performance cost of using reflection but that cost should be negligible compared to the DB operation itself).

I copied and tweaked this approach from https://sii.pl/blog/en/using-table-valued-parameters-with-dapper-in-net/.